x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/runtime_mappings/runtime_mappings.tsx

```diff
@@ -171,15 +175,32 @@ export const RuntimeMappings: FC<Props> = ({ actions, state }) => {
                     <EuiSwitch
                       disabled={jobType === undefined}
                       label={i18n.translate(
-                        'xpack.ml.dataframe.analytics.createWizard.advancedEditorRuntimeMappingsSwitchLabel',
+                        'xpack.ml.dataframe.analytics.createWizard.advancedEditorRuntimeFieldsSwitchLabel',
                         {
-                          defaultMessage: 'Edit runtime mappings',
+                          defaultMessage: 'Edit runtime fields',
                         }
                       )}
                       checked={isRuntimeMappingsEditorEnabled}
-                      onChange={() => toggleEditorHandler()}
+                      onChange={() => {
```

Comment:
I'm finding that the check for un-applied edits isn't made until I first hit the Apply changes button. Same thing happens with the Transforms and DFA wizards.

![switch_modal](https://user-images.githubusercontent.com/7405507/114730782-14a53480-9d39-11eb-8180-aeb16b34b085.gif)

Reply:
I modified so that the behavior of the editors should pop up the modal if the user toggles the switch without previously applying any change. The changes are here [`740e58b` (#97008)](https://github.com/elastic/kibana/pull/97008/commits/740e58bf4ee51fadb639d71de4816bb2c0f2a51e) and here [`a6c7300` (#97008)](https://github.com/elastic/kibana/pull/97008/commits/a6c730024466e9fdcc6b1a8ee0240789324583ce)